src/main/java/com/teamzunior/stringcalculator/Calculation.java

```diff
@@ -0,0 +1,64 @@
+package com.teamzunior.stringcalculator;
+
+
+import com.teamzunior.stringcalculator.operator.Operators;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class Calculation {
+
+    private final Operators registerOperators;
+    private final List<Double> numbers = new ArrayList<>();
+    private final List<String> operators = new ArrayList<>();
+
+    public Calculation(String calculationString, Operators registerOperators) {
+        assert calculationString != null;
+        assert registerOperators != null;
+
+        this.registerOperators = registerOperators;
+
+        initCalculation(calculationString);
+
+        validate();
+    }
+
+    private void initCalculation(String calculationString) {
+        final List<String> values = Arrays.asList(calculationString.split(" "));
+        final int size = values.size();
+
+        //짝수는 Number
+        for (int i = 0; i < size; i = i + 2) {
+            numbers.add(Double.parseDouble(values.get(i)));
+        }
+
+        //홀수는 Operator
+        for (int i = 1; i < size; i = i + 2) {
+            operators.add(registerOperators.parse(values.get(i)));
+        }
+    }
+
+    private void validate() {
+        final int operatorSize = operators.size();
+        final int numberSize = numbers.size();
+
+        //이항연산식은 기본적으로 Operator 개수 * 2 가 Number 의 개수여야한다.
+        if ((operatorSize == 0 && numberSize == 0) || (operatorSize * 2 != numberSize)) {
```

Comment:
조건문이 길어질 경우엔 변수 혹은 함수로 빼는게 더 읽기 좋을 것 같은데 어떻게 생각하시나요?

```suggestion
        if (hasValidSize(operatorSize, numberSize)) {
```